### ISSUE TITLE
Correct `\mathbf` Rendering to Regular Fonts

### ISSUE THREAD

**Issue by xiaodong-hu** [mitex-rs/mitex#142]
Hi, I found that that both `\mathbf .` and `\mathbf{...}` cannot be correctly rendered in desired form as in LaTeX (**bold** AND **regular**). For example, 
```typst
#import "@preview/mitex:0.2.2": *

#mitext(`Reciprocal vector is $\mathbf G$ or $\mathbf{G}$`)
```
gives
![2024-03-20_14-13](https://github.com/mitex-rs/mitex/assets/26814587/86da7f50-64be-40e1-9b66-36d7e40a3844)

But in LaTeX, it should be
![2024-03-20_14-15](https://github.com/mitex-rs/mitex/assets/26814587/9c6a2e84-ac53-4226-852c-e15c599eed79)

> **NOTE: `\bm` (as well as `\boldsymbol`(please also add support for this command)) behaves different from `\mathbf`!**

---

And here is one more tricky thing:

See https://tex.stackexchange.com/questions/294561/using-textbf-vs-mathbf-in-math-mode. Precisely, `\mathbf{...}` and `\textbf{...}` are different in LaTeX, in that **`\mathbf{...}` will remove ALL blank spaces within its arguments**. So probably a more correct rendering is to parse, for example, for the LaTeX code:
```latex
$\sum_{i\in\mathcal I}\mathbf{Op With Space}=\text{blablabla}$
```
![2024-03-20_15-10](https://github.com/mitex-rs/mitex/assets/26814587/4673267e-bf01-4a35-8194-ae78b29f71f4)

as the Typst code:
```typst
$ sum_(i in cal(I))bold(#"Op With Spaces".replace(" ", ""))="blablabla" $ 
```
![2024-03-20_15-18](https://github.com/mitex-rs/mitex/assets/26814587/c12b32c0-3bdd-4977-8400-a82ff6e72de5)

I think this clearly demonstrate how to fix the `\mathbf` rendering problem : )

Thanks for your help!

**Comment by Enter-tainer** [comment]
> in that \mathbf{...} with remove ALL blank spaces within its arguments. 

This is tricky. How about other spaces in unicode? Blindly removing all spaces might be wrong in some cases. I wonder where can we get a spec describing the behavior of mathbf

**Comment by Enter-tainer** [comment]
For now we can first fix the font variant problem and deal with spaces later

**Comment by Enter-tainer** [comment]
https://github.com/mitex-rs/mitex/blob/919336210ad4fa45b4823d302b6819625bdf29c4/packages/mitex/specs/latex/standard.typ#L124

**Comment by Enivex** [comment]
> > in that \mathbf{...} with remove ALL blank spaces within its arguments.
> 
> This is tricky. How about other spaces in unicode? Blindly removing all spaces might be wrong in some cases. I wonder where can we get a spec describing the behavior of mathbf

Math isn't whitespace dependent in LaTeX. The difference here is that `\textbf{}` typesets as regular bold text (which does take whitespace into account, but only a single one if there are consecutive ones), while `\mathbf{}` is for typesetting with bold math symbols.

```
\[
    |\mathbf{ Hello   World }|          |\textbf{ Hello   World }|
\]
```

results in 
![image](https://github.com/mitex-rs/mitex/assets/2191754/27e91d4e-8b3d-4b6e-b519-9f7017984c74)

**Comment by OrangeX4** [comment]
Fixed in 0.3.3
